cli/interactive_tests: re-enable test_sql_mem_monitor

knz opened the pull request:
This test was previously disabled because it leaked root-owned files
caused by distsql temp storage. Actually the root problem was more
serious than that, the test should not have been using distsqsl in the
first place!

This patch corrects the root cause and re-enables the patch.

Also, this test as test_log_flags really *must* use
`start_test`/`end_test`. These two commands also ensure the
directories created by the cockroach process(es) are writable outside
of docker.

By doing so we guarantee that reenabling the test will never cause CI
agents to become hosed with undeletable root-owned files.

(This requirement may be lifted in the future by ensuring
the processes run with the host UID inside docker, but this will be
addressed by a separate patch.)

Supersedes  #17890.
Fixes #17880.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17893)
<!-- Reviewable:end -->

tschottdorf commented:
Wanna add `Fixes #17880`?

LGTM, thanks for fixing!

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/cli/interactive_tests/common.tcl, line 38 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17893#-KsJ_0IvZGYYWP98Jyl4:-KsJ_0IvZGYYWP98Jyl5:bucdcv0) ([raw file](https://github.com/cockroachdb/cockroach/blob/f3b33190325a83d497334fd4954ad29e29caa3aa/pkg/cli/interactive_tests/common.tcl#L38)):*
> ```Tcl
>     # Also CockroachDB doesn't honor umask anyway.
>     # So we simply come after the fact and adjust the permissions.
>     system "find logs -exec chmod a+rw '{}' \\; || true"
> ```

Add a comment here, please.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17893#-:-KsJ_Bv10Is3ghqve7QA:bmyjm5g)*
<!-- Sent from Reviewable.io -->

knz commented:
Done

---

Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/cli/interactive_tests/common.tcl, line 38 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/17893#-KsJ_0IvZGYYWP98Jyl4:-KsJbI8KTZ33xTM7drtV:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f3b33190325a83d497334fd4954ad29e29caa3aa/pkg/cli/interactive_tests/common.tcl#L38)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

Add a comment here, please.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17893#-:-KsJbHCHoshMxXa44SL-:bb6un1j)*
<!-- Sent from Reviewable.io -->

knz committed:
cli/interactive_tests: re-enable test_sql_mem_monitor

This test was previously disabled because it leaked root-owned files
caused by distsql temp storage. Actually the root problem was more
serious than that, the test should not have been using distsqsl in the
first place!

This patch corrects the root cause and re-enables the patch.

Also, this test as test_log_flags really *must* use
`start_test`/`end_test`. These two commands also ensure the
directories created by the cockroach process(es) are writable outside
of docker.

By doing so we guarantee that reenabling the test will never cause CI
agents to become hosed with undeletable root-owned files.

(This requirement may be lifted in the future by ensuring
the processes run with the host UID inside docker, but this will be
addressed by a separate patch.)
